src/main/java/duke/task/Deadlines.java

```diff
@@ -0,0 +1,99 @@
+        }
+        return str;
+    }
+
+    @Override
+    public String toString() {
+        String str;
+        if (this.getIsDone()){
+            str = String.format("[D] %s [X] (by %s)", this.getDescription(),
+                    this.deadline.format(DATE_TIME_OUTPUT_FORMAT));
+        } else {
+            str = String.format("[D] %s [ ] (by %s)", this.getDescription(),
+                    this.deadline.format(DATE_TIME_OUTPUT_FORMAT));
+        }
+        return str;
+    }
+
+//    public duke.task.Deadlines(String input)
+//            throws duke.dukeexception.MissingDescriptionException, duke.dukeexception.MissingDeadlineException, DateTimeParseException {
+//        super();
+//        try {
+//            //remove initial command
+//            String sub = input.substring(9);
+//            int timeIndex = sub.lastIndexOf("/by");
+//            //get description part of input string
+//            if (timeIndex == -1) {
+//                throw new duke.dukeexception.MissingDeadlineException();
+//            }
+//            String description = sub.substring(0, timeIndex - 1);
+//            this.description = description;
+//            String deadlineString = sub.substring(timeIndex + 4);
+//            LocalDateTime deadline = LocalDateTime.parse(deadlineString, DATE_TIME_INPUT_FORMAT);
+//            this.deadline = deadline;
+//        } catch (StringIndexOutOfBoundsException e) {
+//            throw new duke.dukeexception.MissingDescriptionException();
+//        }
+//    }
+//
+//    public duke.task.Deadlines(String input, boolean isDone)
+//            throws duke.dukeexception.MissingDescriptionException, duke.dukeexception.MissingDeadlineException, DateTimeParseException {
+//        super(isDone);
+//        try {
+//            //remove initial command
+//            String sub = input.substring(9);
+//            int timeIndex = sub.lastIndexOf("/by");
+//            //get description part of input string
+//            if (timeIndex == -1) {
+//                throw new duke.dukeexception.MissingDeadlineException();
+//            }
+//            String description = sub.substring(0, timeIndex - 1);
+//            this.description = description;
+//            String deadlineString = sub.substring(timeIndex + 4);
+//            LocalDateTime deadline = LocalDateTime.parse(deadlineString, DATE_TIME_INPUT_FORMAT);
+//            this.deadline = deadline;
+//        } catch (StringIndexOutOfBoundsException e) {
+//            throw new duke.dukeexception.MissingDescriptionException();
+//        }
+//    }
+//
+//    public duke.task.Deadlines(String description, String deadlineString, boolean isDone) throws DateTimeParseException {
+//        super(isDone);
+//        this.description = description;
+//        LocalDateTime deadline = LocalDateTime.parse(deadlineString, DATE_TIME_INPUT_FORMAT);
+//        this.deadline = deadline;
+//    }
+}
```

Comment:
Is there a reason why you didn't delete this chunk of code? It makes your code a bit more messy